### PR TITLE
`Exam mode`: Make participation status change only available after working time is over

### PR DESCRIPTION
**Pull request by florian-glombik**
### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Description
<!-- Describe your changes in detail -->
As noticed in https://github.com/ls1intum/Artemis/pull/7267/files#r1336404731 when trying to change the student exam to submitted while the exam is still running two error
messages are displayed that do not describe the actual problem _(= the submitted state of a student exam cannot be changed while it is in progress)_.

The issue was, that individual working times of students have not been considered for displaying the button.

This PR
* prevents that this error can occur by only making the button available when the exam is finished _(considering the individual working time)_
* introduces the usage of ngElse to reduce error-prone code
* improves the UI by indenting all categories with the same margin left
* show the end of the working time in case an individual working time was set

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
Prerequisites:
- 1 Instructor
- 1 Exam with at least 1 registered student

1. Open the Student exam for the registered student while the exam is running
2. See that `Change Student Exam to Submitted` is not available
3. Wait until the exam is finished _(you can adjust the end date)_
4. See that `Change Student Exam to Submitted` is not available
5. Submit & unsubmit the exam, verify that the translations in the modal are displayed properly
6. Give the student a time extension, which leads to the exam not being finished => press save! 
7. See that the individual exam end date is previewed
8. See that `Change Student Exam to Submitted` is not available anymore

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-------------------:|
| [student-exam-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5555/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/student-exams/student-exam-detail.component.ts.html) | 92.92% | ✅ |


### Screenshots


| Before                                                                                                                                                                                 | After                                                                                                                                                                                                                                                                                                                        |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="1248" alt="not indented consistently and without preview of exam end time" src="https://github.com/ls1intum/Artemis/assets/63976129/8a478ae8-5ae8-4508-b99f-8199247c3035"> | <img width="1248" alt="indented and with individual working time preview" src="https://github.com/ls1intum/Artemis/assets/63976129/a204e849-90e4-42d0-8a89-0bbee757a126"> <img width="1248" alt="exam over button displayed" src="https://github.com/ls1intum/Artemis/assets/63976129/f23e28e6-4b84-4351-8901-70ef0e0b1094"> |

### CODE REVIEW

**Review by tobias-lippert — APPROVED** [review]
Tested in testing session on ts1.

**Review by MaximilianAnzinger — DISMISSED** [review]
For now this works fine.

Followup:
- disable button instead of hiding it
- make the text message bold
- fix indentation of 'no sessions'

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html [hunk 1]
```diff
@@ -139,41 +152,59 @@ <h5><span jhiTranslate="artemisApp.studentExams.participationStatus">Participati
...
                 </li>
             </ol>
+
+            <button
+                class="btn btn-danger"
+                type="button"
+                *ngIf="course?.isAtLeastInstructor"
+                (click)="openConfirmationModal(toggleSubmitPopUp)"
+                [disabled]="!isExamOver() || isSaving"
+                [ngbTooltip]="!isExamOver() || isSaving ? ('artemisApp.studentExamDetail.disabledChangeSubmissionStateButtonExplanation' | artemisTranslate) : ''"
```
Comment (valentin-boehm) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html:162]: Tooltips are not visible on disabled buttons, so this never displayed. One possible workaround might be to wrap the button into a div and have the tooltip on the div.

**Review by valentin-boehm — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html (hunk 1 above)
Comment (florian-glombik) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html]: Thank you for the review and the solution suggestion!

I should have tested it better; it is now adjusted :)

**Review by valentin-boehm — APPROVED** [review]
lgtm

**Review by rstief — APPROVED** [review]
Tested in Testing Session, looks good

**Review by basak-akan — APPROVED** [review]
Tested in testing session on TS1, worked well 👍

**Review by tobias-lippert — APPROVED** [review]
Reapprove

**Pull request merged**